src/paths.jl

```diff
@@ -23,14 +15,32 @@ function path!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianI
 
     for i in 1:length(vertices)-1
         if min(f,vertices[i+1])==f && max(l,vertices[i+1])==l
-            line!(img, vertices[i], vertices[i+1], color)
+            draw!(img, LineSegment(vertices[i], vertices[i+1]), color)
         else
             println(vertices[i+1])
             error("Point coordinates out of range.")
         end
     end
+end
 
-    if closed==true
-        line!(img, vertices[1], vertices[end], color)
-    end
-end
\ No newline at end of file
+#Polygon methods
+
+Polygon(v::Vector{Tuple{Int, Int}}) = Polygon([Point(p...) for p in v])
+Polygon(v::Vector{CartesianIndex{2}}) = Polygon([Point(p) for p in v])
```

Comment:
> Do these result in the same as `Polygon.(v)` ? Should we provide explicit vectorized methods if broadcast would do the trick?

I think you misinterpreted these methods. `Polygon` requires a list of `Point` as a parameter. That list can be passed as either [`Vector{Point}`](https://github.com/JuliaImages/ImageDraw.jl/pull/13/files/94c38d84e47b81d61046c1c87bc3f1072d6029d3#diff-ac236c9f23e3c15683602f303d89d8cdR100) , [`Vector{Tuple{Int, Int}}`](https://github.com/JuliaImages/ImageDraw.jl/pull/13/files/94c38d84e47b81d61046c1c87bc3f1072d6029d3#diff-7473990870b1a3ffad8f07ae087f997dR28) or [`Vector{CartesianIndex{2}}`](https://github.com/JuliaImages/ImageDraw.jl/pull/13/files/94c38d84e47b81d610946c1c87bc3f1072d6029d3#diff-7473990870b1a3ffad8f07ae087f97dR29). A `Polygon` cannot be represented using just a single point so `Polygon.(v)` won't work.

> If we do provide these, dispatching on `AbstractVector` would be a little more flexible

Yes, that would be better :smile: